app/controllers/api/v1/doorkeeper/profiles_controller.rb

```diff
@@ -0,0 +1,10 @@
+class Api::V1::Doorkeeper::ProfilesController < Api::BaseController
+  before_action :doorkeeper_authorize!
+
+  def show
+    profile_info = current_user.as_json
+    profile_info['email'] = current_user.email unless profile_info['email']
```

Comment:
This is pretty weird to see. I'd rather have [`User#payload`](https://github.com/rubygems/rubygems.org/blob/master/app/models/user.rb#L63) (which is way too generically named) take a `hide_email` kwarg with default:

``` ruby
def api_payload(hide_email: hide_email)
  {id: id, handle: handle}.tap do |payload|
    payload[:email] = email unless hide_email
  end
end
```

Then in this controller it would just be:

``` ruby
render json: current_user.as_json(hide_email: false)
```